README.md

```diff
@@ -1,35 +1,105 @@
 
-TODO: Delete this and the text above, and describe your gem
+
+This gem is a Ruby wrapper around `@dcoker`'s [biscuit library](https://github.com/dcoker/biscuit), a multi-region HA key-value store for your AWS infrastructure secrets.
+
+By using this Ruby library, it is easy to integrate into a Ruby/Rails stack.
 
 ## Installation
 
-Add this line to your application's Gemfile:
+- Add this line to your application's Gemfile:
+
+    ```ruby
+    gem 'biscuit'
+    ```
+
+- And then run `bundle`.
+
+- `touch` a yaml file (or multiple for different environments).
+
+## Usage
+
+### Loading K/V pairs into a hash
 
 ```ruby
-gem 'biscuit'
+secrets_file = "some_yaml_file.yaml"
+SECRETS = Biscuit::SecretsDecrypter.new(secrets_file).load
+
+puts SECRETS["some_password"]
+# => "decrypted password"
 ```
 
-And then execute:
+### Loading into ENV Vars
 
-    $ bundle
+If you store config in ENV vars as suggested by the [12 Factor App](https://12factor.net/config), you can load your AWS encrypted secrets into ENV vars like this:
 
-Or install it yourself as:
+```ruby
+secrets_file = "some_yaml_file.yaml"
+Biscuit::SecretsDecrypter.new(secrets_file).load do |key, value|
+  ENV[key] = value
+end
+```
 
-    $ gem install biscuit
+This approach pairs with [dotenv](https://github.com/bkeepers/dotenv) really well - dotenv for test/development, and biscuit for staging/production environments.
 
-## Usage
+#### With Rails
+
+Load your secrets in `application.rb`, between loading Rails/bundler, before the Application config starts:
+
+```ruby
+require "rails/all"
+
+...
 
-TODO: Write usage instructions here
+Bundler.require(*Rails.groups)
+
+...
+
+# Add in your biscuit loading here:
+secrets_file = "#{__dir__}/secrets/#{Rails.env}.yml"
+if File.exist?(secrets_file) # You can also check things like if Rails.env.production?
+  Biscuit::SecretsDecrypter.new(secrets_file).load do |key, value|
+    ENV[key] = value
+  end
+end
+
+...
+
+module MyApp
+  class Application < Rails::Application
+    ....
+```
+
+#### Adding a new key
+
+From the application root, run `biscuit put -f`, followed by the path to the yaml you want to encrypt in, followed by the key, followed by the example.
+
+```bash
+$ biscuit put -f config/secrets/production.yml SECRET_KEY "sensitive value"
+```
+
+#### Getting a key (CLI)
+
+```bash
+$ biscuit export -f config/secrets/production.yml | grep "SECRET_KEY"
+```
 
 ## Development
 
 After checking out the repo, run `bin/setup` to install dependencies. Then, run `bin/console` for an interactive prompt that will allow you to experiment.
 
 To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the version number in `version.rb`, and then run `bundle exec rake release` to create a git tag for the version, push git commits and tags, and push the `.gem` file to [rubygems.org](https://rubygems.org).
 
+## License
+
+MIT.
```

Comment:
Should this be a [link to the MIT license](https://opensource.org/licenses/MIT)?